internal/attestations/attestations.go

```diff
@@ -0,0 +1,165 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package attestations
+
+import (
+	"errors"
+
+	"github.com/gittuf/gittuf/internal/gitinterface"
+	"github.com/gittuf/gittuf/internal/rsl"
+	"github.com/gittuf/gittuf/internal/third_party/go-git"
+	"github.com/gittuf/gittuf/internal/third_party/go-git/plumbing"
+	"github.com/gittuf/gittuf/internal/third_party/go-git/plumbing/filemode"
+	"github.com/gittuf/gittuf/internal/third_party/go-git/plumbing/object"
+)
+
+const (
+	Ref                                  = "refs/gittuf/attestations"
```

Comment:
Non blocking, but I think we should think about how we can get this to play nice with existing git attestations like [`gitsign attest`](https://github.com/sigstore/gitsign/blob/f915adc8d4b3ccc417c322071d3dbb4bbd3bf1da/internal/commands/attest/README.md) 🤔  (I'll take an AI to dig into this more)

Reply:
In particular - we should figure out sooner rather than later on breaking name changes (e.g. `refs/gittuf/attestations` vs `refs/attestations/commits`) as well as breaking storage formats.

Reply:
I see #237 is open to track this!